walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,54 @@
 // limitations under the License.
 
 package com.google.sps;
+import com.google.sps.FindMeetingQuery;
+import com.google.sps.MeetingRequest;
+import com.google.sps.TimeRange;
+import java.util.*;
 
-import java.util.Collection;
 
 public final class FindMeetingQuery {
+    Collection<TimeRange> timeRanges = new ArrayList<>();
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if(request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+        return timeRanges;
+    }
+    if(request.getAttendees ().isEmpty() || request.getDuration() == 0 || events.isEmpty()) {
+        timeRanges.add(TimeRange.WHOLE_DAY);
+        return timeRanges;
+    }
```

Comment:
nice usage of guard clauses. not sure if this was intentional, but it's considered good style to get short-circuiting edge cases out of the way first by literally putting them first in the function. then below the guard clauses the main function body becomes more readable as it can now assume that these edge cases won't be hit.